src/Framework/NativeMethods.cs

```diff
@@ -1337,34 +1340,36 @@ internal static int GetParentProcessId(int processId)
             {
                 // Hold the child process handle open so that children cannot die and restart with a different parent after we've started looking at it.
                 // This way, any handle we pass back is guaranteed to be one of our actual children.
-                SafeProcessHandle childHandle = OpenProcess(eDesiredAccess.PROCESS_QUERY_INFORMATION, false, possibleChildProcess.Id);
-                if (childHandle.IsInvalid)
+                using (SafeProcessHandle childHandle = OpenProcess(eDesiredAccess.PROCESS_QUERY_INFORMATION, false, possibleChildProcess.Id))
```

Comment:
I'm not SURE I understand the comment but it SOUNDS like it was intentionally keeping this handle open past the scope of this `using`, let's look at this closely.

Reply:
suppressed